app/src/main/java/org/oppia/app/shim/IntentFactoryShimImpl.kt

```diff
@@ -0,0 +1,63 @@
+package org.oppia.app.shim
+
+import android.content.Context
+import android.content.Intent
+import androidx.fragment.app.FragmentActivity
+import org.oppia.app.drawer.KEY_NAVIGATION_PROFILE_ID
+import org.oppia.app.home.recentlyplayed.RecentlyPlayedActivity
+import org.oppia.app.profile.ProfileActivity
+import org.oppia.app.topic.TopicActivity
+import javax.inject.Inject
+
+/**
+ * Creates intents for ViewModels in order to avoid ViewModel files directly depending on Activites.
+ * When working on a ViewModel file, developers should refrain from directly referencing Activities
+ * by adding all Intent functionality here.
+ *
+ * Please note that this file is temporary and all functionality will be returned to it's respective
+ * ViewModel once Gradle has been removed.
+ */
+// TODO(#1619): Remove file post-Gradle
+class IntentFactoryShimImpl @Inject constructor() : IntentFactoryShim {
+
+  private val TOPIC_ACTIVITY_TOPIC_ID_ARGUMENT_KEY = "TopicActivity.topic_id"
+  private val TOPIC_ACTIVITY_STORY_ID_ARGUMENT_KEY = "TopicActivity.story_id"
+
+  /** Returns [ProfileActivity] intent for [AdministratorControlsAccountActionsViewModel]. */
+  override fun createProfileActivityIntent(fragment: FragmentActivity): Intent {
+    return Intent(fragment, ProfileActivity::class.java)
+  }
+
+  /**
+   * Creates a [TopicActivity] intent for [PromotedStoryViewModel] and passes necessary string
```

Comment:
For this & other shim: move impl KDocs to the interfaces since that's where people will look for method details.

Reply:
okay!